cmake/SetupCompilers.cmake

```diff
@@ -40,11 +40,30 @@
 #
 ###############################################################################
 
-if (NOT RAJA_ENABLE_CLANG_CUDA)
-  set(CMAKE_CXX_STANDARD 14)
-else()
-  set(CMAKE_CXX_STANDARD 11)
-endif ()
+set(COMPILERS_KNOWN_TO_PALEOLITHIC_CMAKE AppleClang Clang GNU MSVC)
+
+include(CheckCXXCompilerFlag)
+if(RAJA_CXX_STANDARD_FLAG MATCHES default)
+  if("cxx_std_17" IN_LIST CMAKE_CXX_KNOWN_FEATURES)
+    set(CMAKE_CXX_STANDARD 17)
+  elseif("cxx_std_14" IN_LIST CMAKE_CXX_KNOWN_FEATURES)
+    set(CMAKE_CXX_STANDARD 14)
+  elseif("${CMAKE_CXX_COMPILER_ID}" IN_LIST COMPILERS_KNOWN_TO_PALEOLITHIC_CMAKE)
+    set(CMAKE_CXX_STANDARD 14)
+  else() #cmake has no idea what to do, do it ourselves...
+    foreach(flag_var "-std=c++17" "-std=c++1z" "-std=c++14" "-std=c++1y" "-std=c++11")
+      CHECK_CXX_COMPILER_FLAG(${flag_var} COMPILER_SUPPORTS_${flag_var})
+      if(COMPILER_SUPPORTS_${flag_var})
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${flag_var}")
+        break()
+      endif()
+    endforeach(flag_var)
+  endif()
+else(RAJA_CXX_STANDARD_FLAG MATCHES default)
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${RAJA_CXX_STANDARD_FLAG}")
```

Comment:
Might be worth printing a message like:

`Using C++ standard flag: ${RAJA_CXX_STANDARD_FLAG}`

Reply:
👍